File: cheatshield-mobile/cheatshield/lib/models/quiz.dart
import 'package:cheatshield/models/question.dart';

class Quiz {
  final int idQuiz;
  final String title;
  final String code;
  final String description;
  final String publishedAt;
  final String validUntil;
  final int timeLimit;
  final List<Question> questions;

  Quiz({
    required this.idQuiz,
    required this.title,
    required this.code,
    required this.description,
    required this.publishedAt,
    required this.validUntil,
    required this.timeLimit,
    required this.questions,
  });
}

File: cheatshield-admin/app/Models/Quiz.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Quiz extends Model
{
    use HasFactory, HasUuids, SoftDeletes;

    protected $primaryKey = 'id';
    protected $keyType = 'string';
    public $incrementing = false;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'user_id',
        'title',
        'description',
    ];

    /**
     * The attributes that should be cast to native types.
     *
     * @var array
     */
    protected $casts = [
        'published_at' => 'timestamp',
    ];

    /**
     * @return BelongsTo<User,Quiz>
     */
    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    /**
     * @return HasMany<Question,Quiz>
     */
    public function questions(): HasMany
    {
        return $this->hasMany(Question::class);
    }
}
